### DISCUSSION TITLE
Browser-Accessible Version (Docker Container)

### DISCUSSION THREAD

**Discussion by zimmra** [lencx/Noi#122] [Show and tell]
I wanted persistence across my workstations at multiple locations
KasmVNC based docker containers provide an easy way to access GUI-based Linux applications from a web browser

I use this among other Kasm Workspaces, but should function just fine as a standalone container as well

Has been working great for me, feel free to use/fork/criticize at your leisure

https://github.com/zimmra/docker-noi 

![image](https://github.com/lencx/Noi/assets/28514085/ad39ff8e-bf3f-47c5-a2e9-2ebe426d41c5)

**Comment by lencx** [comment]
The work you've done is awesome! I haven't actually tested Noi on Linux, so I wasn't sure if it would work. It's great to hear that you were successful with it.

**Reply by lencx** [reply]
@zimmra The action buttons at the top of the window seem a bit redundant. I can support setting them to a narrow border in the next version, which would be more in line with browser usage.

**Reply by zimmra** [reply]
I am glad you like it @lencx. 

>The action buttons at the top of the window seem a bit redundant.

I wouldn't say they're necessarily redundant, I think it's just inherent when using a 'browser-in-browser' applications, unless putting the 'user' browser in fullscreen there will always be some duplicate buttons between the 'user' browser and the virtual browser. 

With it being a virtual session my main browsers' back/refresh buttons do not interact with Noi browser directly. 

Refreshing reloads the 'virtual session webpage', which should just resume the existing Noi session you are in. 

>I haven't actually tested Noi on Linux, so I wasn't sure if it would work

The only 'real' issue I would say is with Linux/GTK's handling of the GTK menubar and the 'status bar' within he Noi browser.

If the Noi application is not ran in fullscreen on Linux, the GTK menubar shifts the view down within the window, making the Noi 'status bar' inaccessible. This is the case both when the window is maximized or non-maximized. 

The only resolution is to run the application in fullscreen, this hides the GTK menubar and provides room for the Noi 'status bar' to be shown. (My docker container is supposed to do this to the Noi application upon startup, but may not always work)

Here is before/after what it looks like with a regular maximized window versus the fullscreen view

### Maximized (not-fullscreen)
![image](https://github.com/lencx/Noi/assets/28514085/868c8a39-98d6-4442-b06d-2b06ae1f4f1c)

### Fullscreen
![image](https://github.com/lencx/Noi/assets/28514085/bcfccd61-ad6f-4e16-9493-6cf4013eaa38)

**Comment by leohxj** [comment]
amazing!
